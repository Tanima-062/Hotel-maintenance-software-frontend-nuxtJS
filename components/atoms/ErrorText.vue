<template>
  <transition name="fade">
    <p
      v-if="!!errText"
      :class="`text-alert ${NoWrap ? 'whitespace-no-wrap' : ''}`"
    >
      {{ errText }}
    </p>
  </transition>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  errText!: string

  @Prop({ default: false })
  NoWrap!: boolean
}
</script>
<style lang="sass" scoped>
.fade-enter-active, .fade-leave-active
  transition: opacity .2s

.fade-enter, .fade-leave-to /* .fade-leave-active below version 2.1.8 */
  opacity: 0
</style>
