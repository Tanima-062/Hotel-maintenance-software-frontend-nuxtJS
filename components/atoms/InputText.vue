<template>
  <input
    :id="id"
    :type="type"
    class="rounded-md border border-solid text-black-dark placeholder-black-light px-3 focus:shadow-outline focus:outline-none"
    :class="dynamicStyle"
    :placeholder="placeholder"
    :value="value"
    :disabled="disabled"
    @input="input"
    @keydown="isNumber"
    @focus="isFocus = true"
    @blur="blur"
    @keydown.space="(event) => (preventSpace ? event.preventDefault() : event)"
  />
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: null })
  id?: string

  @Prop({ default: 'text' })
  type!: string

  @Prop({ default: '' })
  placeholder!: string

  @Prop({ default: '' })
  value!: string

  @Prop({ default: false })
  error!: boolean

  @Prop({ default: 'bg-white' })
  bg!: string

  @Prop({ default: false })
  disabled!: boolean

  @Prop({ default: 'py-2' })
  padding!: string

  @Prop({ default: false })
  preventSpace!: boolean

  private isFocus: boolean = false

  get dynamicStyle() {
    const classes = []
    // if (this.isFocus) {
    // classes.push('border-blue-600')
    if (this.error) {
      classes.push('border-alert')
      classes.push('bg-error')
    } else {
      classes.push(this.bg)
      classes.push('border-bdColor')
    }
    classes.push(this.padding)
    return classes.join(' ')
  }

  private isNumber(event: Event & { keyCode: number }): boolean {
    if (this.type !== 'number') return true
    const preventKey = [38, 40, 190, 69, 187, 189]
    if (preventKey.includes(event.keyCode)) {
      event.preventDefault()
      return false
    } else if (this.value.length >= Number(this.$attrs.maxlength)) {
      if (event.keyCode !== 8) {
        event.preventDefault()
        return false
      }
      return true
    } else {
      return true
    }
  }

  @Emit()
  private input(e: Event & { target: HTMLInputElement }): string {
    return e.target.value
  }

  @Emit()
  private blur(e: Event & { target: HTMLInputElement }): string {
    this.isFocus = false
    return e.target.value
  }
}
</script>
