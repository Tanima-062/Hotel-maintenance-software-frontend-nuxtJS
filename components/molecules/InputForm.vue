<template>
  <validation-provider
    v-slot="{ errors }"
    tag="div"
    :rules="rules"
    :name="name"
  >
    <label class="text-black-dark mb-2">{{ label }}</label>
    <div class="relative">
      <InputText
        :type="type"
        :class="styles"
        :error="!!errors[0]"
        :placeholder="placeholder"
        :value="value"
        :bg="bg"
        :padding="padding"
        :disabled="disabled"
        :maxlength="textLength"
        :prevent-space="preventSpace"
        @input="input"
        @blur="blur"
      />
      <div
        v-if="showTextLength"
        class="absolute right-0 bottom-0 text-inputLength text-sm py-2.5 pr-2"
      >
        {{ inputTextLength }}/ {{ textLength }}
      </div>
    </div>
    <ErrorText v-if="showError" :no-wrap="errorNoWrap" :err-text="errors[0]" />
  </validation-provider>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component({
  components: {
    InputText: () => import('~/components/atoms/InputText.vue'),
    ErrorText: () => import('~/components/atoms/ErrorText.vue'),
  },
})
export default class extends Vue {
  @Prop({ default: 'text' })
  label!: string

  @Prop({ default: 'text' })
  type!: string

  @Prop({ default: '' })
  placeholder!: string

  @Prop({ default: '' })
  value!: string

  @Prop({ default: '' })
  name!: string

  @Prop({ default: '' })
  rules!: string

  @Prop({ default: '' })
  bg!: string

  @Prop({ default: false })
  disabled!: boolean

  @Prop({ default: false })
  showTextLength!: boolean

  @Prop({ default: 1000 })
  textLength!: number

  @Prop({ default: 'w-full' })
  width!: string

  @Prop({ default: false })
  errorNoWrap!: boolean

  @Prop({ default: true })
  showError!: boolean

  @Prop({ default: 0 })
  inputTextLength!: number

  @Prop({ default: 'py-2' })
  padding!: string

  @Prop({ default: false })
  preventSpace!: boolean

  get styles(): string {
    const style = []
    if (this.showTextLength) {
      style.push('pr-16')
    }
    if (this.type === 'number') {
      style.push('text-right')
    }
    if (this.disabled) {
      style.push('cursor-not-allowed ')
    }
    style.push(this.width)
    return style.join(' ')
  }

  @Emit()
  private input(e: string | number): string | number {
    return e
  }

  @Emit()
  private blur(e: string | number): string | number {
    return e
  }
}
</script>
