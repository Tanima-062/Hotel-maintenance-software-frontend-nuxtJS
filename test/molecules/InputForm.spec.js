import { mount } from '@vue/test-utils'
import InputForm from '@/components/molecules/InputForm.vue'
import InputText from '@/components/atoms/InputText.vue'
import ErrorText from '@/components/atoms/ErrorText.vue'

describe('Molecules InputForm Check', () => {
  test('InputForm Test', () => {
    const wrapper = mount(InputForm, {
      stubs: {
        InputText,
        ErrorText,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
