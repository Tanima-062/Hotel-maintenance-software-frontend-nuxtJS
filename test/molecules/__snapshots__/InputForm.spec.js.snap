// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules InputForm Check InputForm Test 1`] = `
<div>
  <label
    class="text-black-dark mb-2"
  >
    text
  </label>
   
  <div
    class="relative"
  >
    <input
      class="rounded-md border border-solid text-black-dark placeholder-black-light px-3 focus:shadow-outline focus:outline-none  border-bdColor py-2 w-full"
      maxlength="1000"
      placeholder=""
      type="text"
    />
     
    <!---->
  </div>
   
  <!---->
</div>
`;
